blockchain/network.py

```diff
@@ -466,10 +476,25 @@ def phase_3_broadcast(self, block_info, phase_type):
                 logger().warning('failed to submit to node %s', node.node_id)
                 continue
 
-    # TODO: implement this broadcast
-    # TODO: make sure to add in vrs = self.get_vrs(node, ver_ids) self.resolve_data(vrs, 4)
     def phase_4_broadcast(self, block_info, phase_type):
-        pass
+        """ send phase_4 information for phase_5 execution """
+        ver_ids = []
+
+        verification_record = block_info['verification_record']
+        verification_info = verification_record['verification_info']
+
+        phase_4_msg = message_types.Phase_4_msg()
+        phase_4_msg.record = thrift_converter.convert_to_thrift_record(verification_record)
+        phase_4_msg.lower_hash = verification_info
```

Comment:
OK - looked at the other phases, and this is the only one that has the `lower_phase` marked in the network module instead of the processing module `execute_phase_x()` function.

It looks like the phase 4 verification info is just a hash (it's only a notary function) but I want to know why this is set within the network module unlike the other phases.

If you have time this week or weekend, can we do a quick walkthrough?